docs/user-guide/spark-user-guide.md

```diff
@@ -0,0 +1,159 @@
+
+Spark configuration itself can be quite complex and there are a number of configuration options assumed to in place for the above example to function. These properties could be specified as `--conf` arguments to spark-submit or our preference is to store these in a separate properties file such as SPARK_HOME\conf\spark-defaults.conf. This guide makes no attempt to cover the complexity of Spark configuration. Necessary properties will vary based on the deployment environment and application needs. For reference, the full set of Spark properties available can be found [here](https://spark.apache.org/docs/latest/configuration.html#spark-configuration). Details on how to configure Spark to run on Kubernetes can be found [here](https://spark.apache.org/docs/latest/running-on-kubernetes.html). 
+
+Application consumers should, at minimum, pay close attention to the following properties:
+
+* spark.master
+* spark.driver.memory
+* spark.driver.memoryOverhead
+* spark.executor.memory
+* spark.executor.memoryOverhead
+
+Users running Spark under Kubernetes will probably also need at least the following options:
+
+* spark.kubernetes.namespace
+* spark.kubernetes.authenticate.driver.serviceAccountName
+* spark.kubernetes.container.image
+
+#### Configuration-related Options
+
+The Spark CQL Evaluator application reads application configuration data from a number of files that work together to describe the desired program behavior. The content for each of those configuration files will be discussed in detail below, but, in summary, the configuration options are the path to a filesystem folder containing files in either CQL or ELM+XML format (`-c`), the path to a CQL modelinfo file in XML format (`-m`), and paths to a context-definitions.json file that describes how Spark data will be grouped prior to CQL evaluation (`-d`) and a cql-jobs file that describes which CQL libraries and expressions will be executed during evaluation (`-j`). All *configuration data* is assumed to be available in the local filesystem where the Spark application is running. For a Spark cluster running under Kubernetes, this can be done using a Persistent Volume (PV)/Persistent Volume Claim (PVC). In the IBM cloud, we test with a PVC that is backed by a PV that resides in IBM Cloud Object Storage (COS). The IBM Cloud team has an excellent [tutorial](https://www.ibm.com/cloud/blog/kubernetes-persistent-volumes-backed-by-ibm-cloud-object-storage-buckets) that describes how to setup your COS-backed PVC that you should review if you wish to pursue this option.
```

Comment:
Somewhere here it would be helpful to insert some kind of overall architectural diagram.